src/nvim/generators/gen_events.lua

```diff
@@ -16,46 +15,28 @@ names_tgt:write([[
 static const struct event_name {
   size_t len;
   char *name;
-  event_T event;
+  int event;
 } event_names[] = {]])
 
+local aliases = 0
 for i, event in ipairs(events) do
-  enum_tgt:write(('\n  EVENT_%s = %u,'):format(event:upper(), i - 1))
-  names_tgt:write(('\n  {%u, "%s", EVENT_%s},'):format(#event, event, event:upper()))
+  enum_tgt:write(('\n  EVENT_%s = %u,'):format(event[1]:upper(), i + aliases - 1))
+  -- Events with positive keys aren't allowed in 'eventignorewin'.
+  local event_int = ('%sEVENT_%s'):format(event[3] and '-' or '', event[1]:upper())
+  names_tgt:write(('\n  {%u, "%s", %s},'):format(#event[1], event[1], event_int))
+  for _, alias in ipairs(event[2]) do
+    aliases = aliases + 1
+    names_tgt:write(('\n  {%u, "%s", %s},'):format(#alias, alias, event_int))
+    enum_tgt:write(('\n  EVENT_%s = %u,'):format(alias:upper(), i + aliases - 1))
+  end
   if i == #events then -- Last item.
-    enum_tgt:write(('\n  NUM_EVENTS = %u,'):format(i))
+    enum_tgt:write(('\n  NUM_EVENTS = %u,'):format(i + aliases))
   end
 end
 
-for _, v in ipairs(aliases) do
-  local alias = v[1]
-  local event = v[2]
-  names_tgt:write(('\n  {%u, "%s", EVENT_%s},'):format(#alias, alias, event:upper()))
-end
-
-names_tgt:write('\n  {0, NULL, (event_T)0},')
+names_tgt:write('\n  {0, NULL, (event_T)0},\n};\n')
+names_tgt:write('\nstatic AutoCmdVec autocmds[NUM_EVENTS] = { KV_INITIAL_VALUE };\n')
```

Comment:
This doesn't make sense. It only initializes the first element of `autocmds[]` to `KV_INITIAL_VALUE`, although `KV_INITIAL_VALUE` is all zero anyway...

Reply:
Hmm, or just change it to `0`?